cmake/LLVM.cmake

```diff
@@ -397,23 +401,37 @@ endmacro()
 
 # clang++ try-compile macro
 macro(custom_try_compile_clangxx SOURCE1 SOURCE2 RES_VAR)
-  custom_try_compile_c_cxx("${CLANGXX}" "cc" "${SOURCE1}" "${SOURCE2}" ${RES_VAR}  "-c" ${ARGN})
+  string(RANDOM RNDNAME)
+  set(RANDOM_FILENAME "${CMAKE_BINARY_DIR}/compile_test_${RNDNAME}.o")
+  custom_try_compile_c_cxx("${CLANGXX}" "cc" "${SOURCE1}" "${SOURCE2}" ${RES_VAR} "-o" "${RANDOM_FILENAME}" "-c" ${ARGN})
+  file(REMOVE "${RANDOM_FILENAME}")
 endmacro()
 
 # clang++ try-compile macro
 macro(custom_try_compile_clang SOURCE1 SOURCE2 RES_VAR)
-  custom_try_compile_c_cxx("${CLANG}" "c" "${SOURCE1}" "${SOURCE2}" ${RES_VAR}  "-c" ${ARGN})
+  string(RANDOM RNDNAME)
+  set(RANDOM_FILENAME "${CMAKE_BINARY_DIR}/compile_test_${RNDNAME}.o")
+  custom_try_compile_c_cxx("${CLANG}" "c" "${SOURCE1}" "${SOURCE2}" ${RES_VAR} "-o" "${RANDOM_FILENAME}" "-c" ${ARGN})
+  file(REMOVE "${RANDOM_FILENAME}")
 endmacro()
 
 # clang++ try-compile macro
 macro(custom_try_compile_clang_silent SOURCE1 SOURCE2 RES_VAR)
-  custom_try_compile_c_cxx_silent("${CLANG}" "c" "${SOURCE1}" "${SOURCE2}" ${RES_VAR} "-c" ${ARGN})
+  string(RANDOM RNDNAME)
+  set(RANDOM_FILENAME "${CMAKE_BINARY_DIR}/compile_test_${RNDNAME}.o")
+  custom_try_compile_c_cxx_silent("${CLANG}" "c" "${SOURCE1}" "${SOURCE2}" ${RES_VAR} "-o" "${RANDOM_FILENAME}" "-c" ${ARGN})
+  file(REMOVE "${RANDOM_FILENAME}")
 endmacro()
 
 # clang++ try-link macro
 macro(custom_try_link_clang SOURCE1 SOURCE2 RES_VAR)
-  set(RANDOM_FILENAME "${CMAKE_BINARY_DIR}/compile_test_${RNDNAME}.${SUFFIX}")
-  custom_try_compile_c_cxx_silent("${CLANG}" "c" "${SOURCE1}" "${SOURCE2}" ${RES_VAR}  "-o" "${RANDOM_FILENAME}" ${ARGN})
+  string(RANDOM RNDNAME)
+  if(WIN32)
+    set(RANDOM_FILENAME "${CMAKE_BINARY_DIR}/compile_test_${RNDNAME}.exe")
```

Comment:
Why [${SUFFIX}](https://cmake.org/cmake/help/v3.0/prop_tgt/SUFFIX.html) does not work here? Perhaps the problem here is that it should be used as `... ${RNDNAME}${SUFFIX}` (no dot).

Reply:
`SUFFIX` is not an argument to this macro, which it is with other macros (like `custom_try_compile_c_cxx_silent` or `custom_try_compile_c_cxx`). I don't see how the linked target property is relevant here?

Reply:
I thought the SUFFIX was a predefined CMake variable and didn't read the description carefully.

Probably the SUFFIX was mixed up with CMAKE_EXECUTABLE_SUFFIX. Could you use the latter?